### PR TITLE
Feature/fileitem factory

### PR DESCRIPTION
**Pull request by Toflar**
Replaces https://github.com/terminal42/contao-notification_center/pull/360.

### CODE REVIEW

File: config/services.php [hunk 1]
```diff
@@ -59,6 +60,12 @@
         ])
     ;
 
+    $services->set(FileItemFactory::class)
+        ->args([
+            service('mime_types'),
```
Comment (fritzmg) [inline config/services.php:65]: I think this needs a

```suggestion
            service('mime_types')->nullOnInvalid(),
```

as the `mime_types` service is defined by the `symfony/framework-bundle` and thus might not exist. Symfony itself also uses `->args([service('mime_types')->nullOnInvalid()])` in its various components.

**Comment by Toflar** [comment]
Is that what you need @aschempp? If so, I'll merge this :)

**Review by fritzmg — APPROVED** [review]

**Review by aschempp — COMMENTED** [review]
Looks good except for my notes :)

File: src/BulkyItem/FileItemFactory.php [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+use Terminal42\NotificationCenterBundle\Exception\BulkyItem\InvalidFileItemException;
+
+class FileItemFactory
+{
+    public function __construct(private readonly MimeTypeGuesserInterface|null $mimeTypeGuesser = null)
+    {
+    }
+
+    /**
+     * @throws InvalidFileItemException if the information cannot be fetched automatically (e.g. missing mime type guesser service)
```
[the remark this answers is not in the document]
Comment (Toflar) [reply src/BulkyItem/FileItemFactory.php:20]: Both comments are correct, you would need to merge them in your suggestion.

File: src/BulkyItem/FileItemFactory.php [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+    }
+
+    /**
+     * @throws InvalidFileItemException
+     */
+    public function createFromVfsFilesystemItem(FilesystemItem $file, VirtualFilesystemInterface $virtualFilesystem): FileItem
+    {
+        $stream = $virtualFilesystem->readStream($file->getPath());
+
+        return FileItem::fromStream($stream, $file->getName(), $file->getMimeType(), $file->getFileSize());
```
[the remark this answers is not in the document]
Comment (Toflar) [reply src/BulkyItem/FileItemFactory.php:42]: It does whatever the implementation of the virtualfilesystem does. We don't care here.

File: src/BulkyItem/FileItemFactory.php [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+    {
+    }
+
+    /**
+     * @throws InvalidFileItemException if the information cannot be fetched automatically (e.g. missing mime type guesser service)
+     */
+    public function createFromLocalPath(string $path): FileItem
+    {
+        if (!(new Filesystem())->exists($path)) {
+            throw new InvalidFileItemException(\sprintf('The file "%s" does not exist.', $path));
```
[the remark this answers is not in the document]
Comment (Toflar) [reply src/BulkyItem/FileItemFactory.php:25]: No, it's best practice to use domain exceptions.

**Pull request merged**